pkg/analyze/data_source.go

```diff
@@ -114,3 +118,67 @@ func NewDataSourceFromClients(
 		ScyllaClusterLister:  scyllaClusterLister,
 	}, nil
 }
+
+func NewDataSourceFromArchive(
+	ctx context.Context,
+	archivePath string,
+) (*DataSource, error) {
+
+	podType := reflect.TypeOf(&corev1.Pod{})
+	serviceType := reflect.TypeOf(&corev1.Service{})
+	secretType := reflect.TypeOf(&corev1.Secret{})
+	configMapType := reflect.TypeOf(&corev1.ConfigMap{})
+	serviceAccountType := reflect.TypeOf(&corev1.ServiceAccount{})
+	scyllaClusterType := reflect.TypeOf(&scyllav1.ScyllaCluster{})
+
+	indexers, err := IndexersFromArchive(os.DirFS(archivePath))
+	if err != nil {
+		return nil, err
+	}
+
+	podIndexer, ok := indexers[podType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for Pods")
+	}
+	podLister := corev1listers.NewPodLister(podIndexer)
+
+	serviceIndexer, ok := indexers[serviceType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for Services")
+	}
+	serviceLister := corev1listers.NewServiceLister(serviceIndexer)
+
+	secretIndexer, ok := indexers[secretType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for Secrets")
+	}
+	secretLister := corev1listers.NewSecretLister(secretIndexer)
+
+	configMapIndexer, ok := indexers[configMapType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for ConfigMaps")
+	}
+	configMapLister := corev1listers.NewConfigMapLister(configMapIndexer)
+
+	serviceAccountIndexer, ok := indexers[serviceAccountType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for ServiceAccounts")
+	}
+	serviceAccountLister := corev1listers.NewServiceAccountLister(serviceAccountIndexer)
+
+	scyllaClusterIndexer, ok := indexers[scyllaClusterType]
+	if !ok {
+		return nil, fmt.Errorf("no indexer provided for ScyllaClusters")
+	}
+	scyllaClusterLister := scyllav1listers.NewScyllaClusterLister(scyllaClusterIndexer)
```

Comment:
what if either of those types had no objects in the archive? E.g. you can easily have a valid must-gather with no scylla clusters

Reply:
the error messages are also quite misleading - you are the one creating the indexers